homeassistant/components/spotify/config_flow.py

```diff
@@ -34,6 +34,14 @@ def extra_authorize_data(self) -> dict:
             "user-read-playback-state",
             # Needed to determine if the user has Spotify Premium
             "user-read-private",
+            # Needed for media browsing
+            "playlist-read-private",
+            "playlist-read-collaborative",
+            "user-library-read",
+            "user-top-read",
+            "user-read-playback-position",
+            "user-read-recently-played",
+            "user-follow-read",
```

Comment:
We need to check and trigger re-auth if current users don't have these scopes yet.

Reply:
Right. This has already been discussed https://github.com/home-assistant/core/pull/39240#discussion_r477183064

Reply:
Rebased